### DISCUSSION TITLE
Dynamic select

### DISCUSSION THREAD

**Discussion by vit1251** [vlang/v#11790] [Ideas]
How about dynamic selection:

```
s:= select
s.push(socket1)
s.push(socket2)
s.push(chan1)
s.push(event)
socks := s.wait()
```

or even some like libev event loop for async developers?

**Comment by vadymzw** [comment]
I like the idea to dynamically change list of select-objects! Also would be helpful to have ability to wait here for not-channel sync primitives.

**Comment by hholst80** [comment]
The dynamic select is already possible for certain classes of "awaitable" types, such as [Channels](https://modules.vlang.io/sync.html#channel_select)

Select on a static set of awaitables is already implemented in [channel select](https://github.com/vlang/v/blob/master/doc/docs.md#channel-select)

**Reply by vadymzw** [reply]
There is no problem with static set.
Ok, let's imagine semi-abstract example. Server listens for connections. Have some functionality that requires to create "awaitable" channel per received connection. Have also single coroutine that "select" for all created channels to do something on every "selected" channel event. What I want to do is appending channel to "select" channels on new connection arrived and drop corresponding channel if connection closed. And prefer not to restart "select" on all existing channels while adding or removing. As you can see this "select" should allow to add and remove its "awaitables" while awaiting on them.
Hope it is clear now..
